corehq/apps/events/models.py

```diff
@@ -478,6 +481,25 @@ def save(self, *args, **kwargs):
         helper.update(case_data)
 
 
+def _parse_id_list_str(locations_str):
+    """
+    Given a string representation of a list of IDs, returns a list.
+
+    >>> _parse_id_list_str("['abc123', 'def456']")
```

Comment:
It feels weird that we have to do this?

Reply:
I wonder if this doesn't have to do with the fact that `locations` on the model [is a TextField](https://github.com/dimagi/commcare-hq/blob/b7ea6b5ddc846dc1d69fb360e42a657ccf26e63c/corehq/apps/events/models.py#L437), although I haven't looked too deeply into it so can't say for certain.

Reply:
@Charl1996 I think that's exactly why. I wanted to use an `ArrayField` with `CharField` items, but that broke the widget. We don't use `ArrayField` with mobile workers either, we use `CharField`, so I went with `TextField` because it doesn't need you to specify `max_length`.